File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        expectedSum = (len(nums)/2)*(len(nums) + 1)
        actualSum = 0
        i = 0

        while(i < len(nums)):
            actualSum += nums[i]
            i += 1
        return int(expectedSum - actualSum)

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums ){
        sort(nums);
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i) {
                return i;
            }
        }return nums.length;
    }
    
    static int[] sort(int[] nums) {
       int i = 0;
       while (i < nums.length) {
        if(nums[i] != i && nums[i] < nums.length){
                swap(nums, i, nums[i]);
            }
            else{
                i++;
            }

       }
       return nums;
        
    }

    static void swap(int[] nums, int numOne, int numTwo){
        int temp = nums[numOne];
        nums[numOne] = nums[numTwo];
        nums[numTwo] = temp;
    }
}